src/controllers/snippet.controller.ts

```diff
@@ -55,3 +64,56 @@ export async function getSnippets(req: Request, res: Response) {
     });
   }
 }
+
+export async function shareSnippet(req: Request, res: Response) {
+  try {
+    const { share, email, snippetid, user_name } = req.body;
+    logger.info(`REQ : Share snippet => ${snippetid}`);
+
+    const id = encodeURIComponent(encrypt(snippetid));
+
+    logger.info(`encrypted mongoDB id => ${id}`);
+
+    if (share) {
+      //update the db, that this particular snippet is sharable
+      const snippet = await UPDATE_SNIPPET_SHARE_STATUS(snippetid);
+      if (snippet) {
+        const url = `https://snippsavvy.com/collab?snippet=${id}?sharing=true`;
+        logger.info(`snippet sharing url generated => ${url}`);
+        return res.status(200).json({ url: url });
+      }
+    } else {
+      if (!email) {
+        return res.status(500).json({ msg: "BAD REQUEST" });
+      }
+      // write logic for sending link in a mail
+      const newemail = encodeURIComponent(encrypt(email));
+      const snippet = SHARE_SNIPPET_PERSONALLY(snippetid, email);
+      const url = `https://snippsavvy.com/collab?snippet=${id}?email=${newemail}`;
```

Comment:
Can we move this url to somewhere else??

Reply:
nope, url is getting created here only

Reply:
ok